src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -33,39 +39,50 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
         }
 
         this._braintreeSDKCreator.initialize(paymentMethod.clientToken);
+        const container = `#${options.containerId}`;
+
+        this._renderButtonsData = {
+            paymentMethod,
+            paypalOptions,
+            container,
+        };
 
         return Promise.all([
-            this._braintreeSDKCreator.getPaypalCheckout(),
-            this._paypalScriptLoader.loadPaypal(),
+            this._braintreeSDKCreator.getPaypalCheckout(() => this.renderButtons()),
+            this._braintreeSDKCreator.getPaypal(),
         ])
-            .then(([paypalCheckout, paypal]) => {
+            .then(([paypalCheckout]) => {
                 this._paypalCheckout = paypalCheckout;
+            });
+    }
 
-                const allowedSources = [];
-                const disallowedSources = [];
+    renderButtons() {
+        const allowedSources = [];
+        const disallowedSources = [];
+        const { paypalOptions, paymentMethod, container } = this._renderButtonsData as RenderButtonsData;
 
-                if (paypalOptions.allowCredit) {
-                    allowedSources.push(paypal.FUNDING.CREDIT);
-                } else {
-                    disallowedSources.push(paypal.FUNDING.CREDIT);
-                }
-
-                return paypal.Button.render({
-                    env: paymentMethod.config.testMode ? 'sandbox' : 'production',
-                    commit: paypalOptions.shouldProcessPayment ? true : false,
-                    funding: {
-                        allowed: allowedSources,
-                        disallowed: disallowedSources,
-                    },
-                    style: {
-                        shape: 'rect',
-                        label: this._offerCredit ? 'credit' : undefined,
-                        ...pick(paypalOptions.style, 'layout', 'size', 'color', 'label', 'shape', 'tagline', 'fundingicons'),
-                    },
-                    payment: () => this._setupPayment(paypalOptions.shippingAddress, paypalOptions.onPaymentError),
-                    onAuthorize: data => this._tokenizePayment(data, paypalOptions.shouldProcessPayment, paypalOptions.onAuthorizeError),
-                }, options.containerId);
-            });
+        if (paypalOptions.allowCredit) {
+            allowedSources.push(this._window.paypal?.FUNDING.CREDIT);
+        } else {
+            disallowedSources.push(this._window.paypal?.FUNDING.CREDIT);
+        }
+        if (this._window && this._window.paypal) {
```

Comment:
Instead of changing `allowedSources` from `string[]` to `Array<string | undefined>`, what you should do is to fix the code itself. i.e.:

```ts
const { paypal } = this._window;

if (paypal) {
    if (paypalOptions.allowCredit) {
        allowedSources.push(paypal.FUNDING.CREDIT);
    } else {
        disallowedSources.push(paypal.FUNDING.CREDIT);
    }

    this._window.paypal.Buttons({
        // ...
    });
}
```

Reply:
fixed